devp2p/kademlia.py

```diff
@@ -52,6 +53,12 @@ def __init__(self, pubkey):
         else:
             assert k_id_size == 256
             self.id = big_endian_to_int(sha3(pubkey))
+        self.ping_recv = False
+        self.pong_recv = False
+
+    @property
+    def bonded(self):
+        return self.ping_recv and self.pong_recv
```

Comment:
something to add here: as far as I know, `geth` solves the out-of-table state tracking by persisting node state in a db. we don't have node persistence available. question is, if it was worth adding?